src/tools/compiletest/src/runtest.rs

```diff
@@ -3049,7 +3061,17 @@ impl<'test> TestCx<'test> {
 
         if self.should_run_successfully() {
             let proc_res = self.exec_compiled_test();
-
+            let run_output_errors = if self.props.check_run_results {
+                self.load_compare_outputs(&proc_res, explicit)
```

Comment:
AFAICT, this means that we'll only compare the final stdout/stderr -- not the compiler output. Can we compare both?

Reply:
It's definitely doable, the only issue is differentiating between expected compiler output and run output. I just pushed some changes that allow comparing both, so let me know if the approach looks okay.